added validation in TeacherDTO & SubjectDTO, fixed error msg in PostM…

osman-jft opened the pull request:
…apping in TeacherController, fixed formatting of code

osman-jft committed:
added validation in TeacherDTO & SubjectDTO, fixed error msg in PostMapping in TeacherController, fixed formatting of code

jaineetsingh-jft commented on src/main/resources/application.properties:
remove this file

osman-jft replied:
done

jaineetsingh-jft commented on src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java:
```diff
@@ -15,26 +17,33 @@ public class TeacherController {
-        //returns list of students from StudentService
+        List<TeacherDTO> teacherDto = teacherService.getAllTeachers();
+        if (teacherDto == null) { // check null condition for list
+            return ResponseDTO.<List<TeacherDTO>>builder()
```
Dont't return response DTO directly from here,throw custom exception

osman-jft replied:
done

jaineetsingh-jft commented on src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java:
```diff
@@ -6,40 +6,40 @@
         List<Student> students = studentRepository.findAll();
-        if(students.isEmpty()){
+        if (students.isEmpty()) {
             return null;
```
throw custom exception

osman-jft replied:
done

osman-jft committed:
trail run for 'add -p' git command

osman-jft committed:
"trying to add & commit via IntelliJ Git GUI"

osman-jft committed:
merged with develop

jaineetsingh-jft commented on src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java:
```diff
@@ -22,32 +20,26 @@ public class StudentController {
-            return ResponseDTO.<List<StudentDTO>>builder()
-                    .data(null).message(Constants.EMPTY_LIST).status(HttpStatus.NO_CONTENT.value())
-                    .build();
+        if (studentDto == null) { // check null condition for list
```
throw exception and handle in service class

osman-jft replied:
done

jaineetsingh-jft commented on src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java:
```diff
@@ -15,26 +17,28 @@ public class TeacherController {
-        return ResponseDTO.<List<TeacherDTO>>builder().data(teacherService.getAllTeachers())
-                .message(Constants.SUCCESS_MSG).status(HttpStatus.OK.value()).build();
+        List<TeacherDTO> teacherDto = teacherService.getAllTeachers();
+        if (teacherDto == null) { // check null condition for list
```
throw exception and hanlde in service class

osman-jft replied:
done

jaineetsingh-jft commented on src/main/resources/application.properties:
remove this file

osman-jft replied:
done

osman-jft committed:
Merge branch 'develop' of github.com:osman-jft/SB_Assign_16_05 into feature-osman

# Conflicts:
#	src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java
#	src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java
#	src/main/java/com/example/sb_assign_16_05_23/dto/ValidList.java
#	src/main/java/com/example/sb_assign_16_05_23/errors/GlobalExceptionHandling.java
#	src/main/java/com/example/sb_assign_16_05_23/service/TeacherService.java
#	src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java
#	src/main/java/com/example/sb_assign_16_05_23/service/impl/TeacherServiceImpl.java
#	src/main/resources/application.properties

osman-jft committed:
Made Requested Changes;
1. Added Custom ExpentionHandling in Service in TeacherServiceImpl
2. Added SwaggerUI functionality in the app; updated pom.xml

jaineetsingh-jft commented on src/main/resources/application.properties:
Delete this file

osman-jft committed:
Made Requested Changes;
1. Deleted application.properties

osman-jft committed:
Merge branch 'develop' of github.com:osman-jft/SB_Assign_16_05 into feature-osman

# Conflicts:
#	pom.xml

osman-jft committed:
Resolved conflicts;
Changes in pom.xml
